cmd/ndm_daemonset/app/command/start.go

```diff
@@ -54,8 +70,49 @@ func NewCmdStart() *cobra.Command {
 			// Start starts registering of probes present in RegisteredProbes
 			probe.Start(probe.RegisteredProbes)
 			ctrl.Start()
+
 		},
 	}
 
 	return getCmd
 }
+
+//GRPCServer starts the grpc server
+func GRPCServer() {
+	{
+
+		// Creating a grpc server, use WithInsecure to allow http connections
+		gs := grpc.NewServer()
+
+		// Creates an instance of Info
+		is := services.NewInfo()
+
+		// Creates an instance of Service
+		ss := services.NewService()
+
+		// Creates an instance of Node
+		ns := services.NewNode()
+
+		// This helps clients determine which services are available to call
+		reflection.Register(gs)
+
+		// Similar to registring handlers for http
+		protos.RegisterInfoServer(gs, is)
+
+		protos.RegisterISCSIServer(gs, ss)
+
+		protos.RegisterNodeServer(gs, ns)
+
+		l, err := net.Listen("tcp", "0.0.0.0:9090")
+		if err != nil {
+			klog.Errorf("Unable to listen %f", err)
+			os.Exit(1)
+		}
+
+		// Listen for requests
+		klog.Info("Starting server at 9090")
+		gs.Serve(l)
+
+	}
+
+}
```

Comment:
Thinking out loud. This function can be made to a separate package so that we can keep the `command` package clean.

The new package can be `cmd/ndm_daemonset/grpcserver`

Reply:
Makes sense.

Reply:
Method can be Start() . So from this file, we call grpcserver.Start()